inline-c-cpp/test/tests.hs

```diff
@@ -275,16 +297,16 @@ main = Hspec.hspec $ do
 tag :: C.CppException -> String
 tag (C.CppStdException {}) = "CppStdException"
 tag (C.CppHaskellException {}) = "CppHaskellException"
-tag (C.CppOtherException {}) = "CppStdException"
+tag (Legacy.CppOtherException {}) = "CppStdException"
 
 shouldBeCppStdException :: Either C.CppException a -> String -> IO ()
-shouldBeCppStdException (Left (C.CppStdException actualMsg)) expectedMsg = do
+shouldBeCppStdException (Left (Legacy.CppStdException actualMsg)) expectedMsg = do
```

Comment:
Something seems wrong with this code. At least on current `master` (commit ef87fbae38ed9f646b912f94606d895d0582f1b4) this pattern always triggers, and the next (non-Legacy) pattern below is dead code in `stack test inline-c-cpp`.

I suspect some confusion with pattern synonyms is going on.

Reply:
It was not a bug, just confusing that the test only covered the legacy API.

Cleared up in my propsed PR https://github.com/fpco/inline-c/pull/154